xcm/xcm-executor/src/traits/matches_fungibles.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use xcm::v0::{MultiAsset, Error as XcmError};
+use sp_std::result;
+
+/// Errors associated with [`MatchesFungibles`] operation.
+pub enum Error {
```

Comment:
This error type cannot (as-is) and should not accommodate all the variant errors that can happen in different implementations (elsewhere), if they can exist. 

If ^^ is correct then we need `type Error` on `MatchesFungibles`, or a `Other(..)` variant for random stuff. 

But for now it seemed like an overkill.